File: Get Started with C Language/6. Function/01_function.c
/*
Basics is same as C++.
It's in my notebook.

Applications of Function/Methods;
- Avoid code redundancy and ease maintenance
- Make code modular
- Abstraction
- Avoid variable name collisions

Prototype Declaration

How function works?
- Function call stack

Inline Function

This are in my notebook.
*/
#include <stdio.h>

void fun(){
	printf("Inside function\n");
}

inline int getMax(int x, int y){
	return (x>y) ? x : y;
}

int main(){
	printf("Before function call\n");
	fun();
	printf("After function call\n");
	return 0;
}

File: Python Programming Foundation/07. Functions/01_function.py
def fun():
    print("fun() called")  # it will be printed when fun is called

def sum(a, b):
    return a+b

print("before calling fun()")
fun()  # fun() is called
fun()  # fun() is called
print(sum(5, 6))
print("after calling fun()")

"""
Applications of Funtions(or Methods)
- Avoid Code Redundancy and easy Maintainance
- Make Code Modular
- Abstraction 
- Avoid varialbe name collisons

"""